lib/scala/filewatcher/src/main/java/org/enso/filewatcher/DefaultWatcher.java

```diff
@@ -0,0 +1,176 @@
+package org.enso.filewatcher;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executor;
+import java.util.function.Consumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Default directory watcher using JDK's {@link WatchService}. */
+public final class DefaultWatcher implements Watcher {
```

Comment:
```suggestion
final class DefaultWatcher implements Watcher {
```
Why should implementation of `Watcher` interface by public?

Reply:
It should not. Refactored in f9d54e3f54f3786e92b757a7ad5089bb9ac5bf51